### PR TITLE
tests/periph_spi: correct test application

### PR DESCRIPTION
**Pull request by PeterKietzmann**
This PR corrects the faulty handover from the function input, to set up the spi `MODE` and `SPEED`. Also it intruduces some `return`s after negative initializations

### CODE REVIEW

File: tests/periph_spi/main.c [hunk 1]
```diff
@@ -91,14 +91,14 @@ int parse_spi_dev(int argc, char **argv)
         return -1;
     }
     if (argc >= 4) {
```
Comment (fnack) [inline tests/periph_spi/main.c]: Is this after the check in line 60 still necessary?

File: tests/periph_spi/main.c [hunk 2]
```diff
@@ -91,14 +91,14 @@ int parse_spi_dev(int argc, char **argv)
...
-        spi_mode = argv[2][0] - '0';
+        spi_mode = argv[3][0] - '0';
         if (spi_mode < 0 || spi_mode > 3) {
             puts("error: invalid MODE value given");
             return -2;
         }
     }
     if (argc >= 5) {
```
Comment (fnack) [inline tests/periph_spi/main.c]: Is this after the check in line 60 still necessary?

File: tests/periph_spi/main.c [hunk 3]
```diff
@@ -91,14 +91,14 @@ int parse_spi_dev(int argc, char **argv)
...
-        spi_speed = argv[3][0] - '0';
+        spi_speed = argv[4][0] - '0';
```
Comment (fnack) [inline tests/periph_spi/main.c]: Couldn't `atoi(argv[4])` be used here as it is for the spi device and cs pin?

File: tests/periph_spi/main.c [hunk 4]
```diff
@@ -91,14 +91,14 @@ int parse_spi_dev(int argc, char **argv)
         return -1;
     }
     if (argc >= 4) {
-        spi_mode = argv[2][0] - '0';
+        spi_mode = argv[3][0] - '0';
```
Comment (fnack) [inline tests/periph_spi/main.c]: Couldn't `atoi(argv[3])` be used here as it is for the spi device and cs pin?

File: tests/periph_spi/main.c (hunk 1 above)
Comment (PeterKietzmann) [inline tests/periph_spi/main.c]: Yes. If `argc` is "just" `>=4`  then "just" set `spi_mode` manually. If it's also `>=5` then also set `spi_speed` manually

File: tests/periph_spi/main.c (hunk 4 above)
Comment (PeterKietzmann) [inline tests/periph_spi/main.c]: Yes I think so. I tried to change as little as possible from the original code. Does this line hurt you?

File: tests/periph_spi/main.c (hunk 2 above)
Comment (PeterKietzmann) [inline tests/periph_spi/main.c]: same as above

File: tests/periph_spi/main.c (hunk 3 above)
Comment (PeterKietzmann) [inline tests/periph_spi/main.c]: same as above

File: tests/periph_spi/main.c (hunk 1 above)
Comment (fnack) [inline tests/periph_spi/main.c]: Ah sorry, I overlooked the default settings at the beginning of the function.

File: tests/periph_spi/main.c (hunk 4 above)
Comment (fnack) [inline tests/periph_spi/main.c]: No, can stay.

**Comment by fnack** [comment]
ACK

**Comment by PeterKietzmann** [comment]
And go

**Pull request merged**